Utilities/DavixAdaptor/src/DavixFile.cc

```
@@ -0,0 +1,346 @@
+  if (!ucert.empty() && getenv("X509_USER_KEY")) {
+    edm::LogInfo("DavixFile") << "X509_USER_{CERT|KEY} found in environment"
+                              << " Will use it for authentication";
+    ukey = getenv("X509_USER_KEY");
+  }
+  // Check if vars are set...
+  if (ucert.empty() || ukey.empty()) {
+    edm::LogWarning("DavixFile") << "Was not able to find proxy in $X509_USER_PROXY, "
+                                 << "X509_USER_{CERT|KEY} or default proxy creation location. "
+                                 << "Will try without authentication";
+    return -1;
+  }
+  return cert->loadFromFilePEM(ukey, ucert, "", davixErr);
+}
+
+void DavixFile::create(const char *name, bool exclusive /* = false */, int perms /* = 066 */) {
+  open(name, (IOFlags::OpenCreate | IOFlags::OpenWrite | IOFlags::OpenTruncate |
+              (exclusive ? IOFlags::OpenExclusive : 0)),
+       perms);
+}
+
+void DavixFile::create(const std::string &name, bool exclusive /* = false */,
+                       int perms /* = 066 */) {
+  open(name.c_str(), (IOFlags::OpenCreate | IOFlags::OpenWrite | IOFlags::OpenTruncate |
+                      (exclusive ? IOFlags::OpenExclusive : 0)),
+       perms);
+}
+
+void DavixFile::open(const std::string &name, int flags /* = IOFlags::OpenRead */,
+                     int perms /* = 066 */) {
+  open(name.c_str(), flags, perms);
+}
+
+void DavixFile::open(const char *name, int flags /* = IOFlags::OpenRead */, int perms /* = 066 */) {
+  // Actual open
+  if ((name == 0) || (*name == 0)) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Cannot open a file without name";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+
+  if ((flags & IOFlags::OpenRead) == 0) {
+    edm::Exception ex(edm::errors::FileOpenError);
+    ex << "Must open file '" << name << "' at least for read";
+    ex.addContext("Calling DavixFile::open()");
+    throw ex;
+  }
+
+  if (davixPosix && m_fd)
```

Calling `open()` on an open file should cause an exception, not silently ignored.